Modules/Filtering/ImageGrid/wrapping/test/OrientImageFilterTest.py

```diff
@@ -27,7 +27,7 @@
 reoriented_image = itk.orient_image_filter(
     image,
     use_image_direction=True,
-    desired_coordinate_orientation=itk.SpatialOrientationEnums.ValidCoordinateOrientations_ITK_COORDINATE_ORIENTATION_LPI
+    desired_coordinate_orientation=itk.AnatomicalOrientation.PositiveEnum.RAS
```

Comment:
Or I might be referring to the enum in correctly... Should this be `itk.AnatomicalOrientation.PositiveEnum_RAS` instead?

Reply:
@blowekamp When building locally, I get the following warnings in the build:

```
[3360/6065] Building CXX object Modules/Filtering/ImageGrid/test/CMakeFiles/ITKImageGridTestDriver.dir/itkOrientImageFilterTest.cxx.o
/home/matt/src/ITK/Modules/Filtering/ImageGrid/test/itkOrientImageFilterTest.cxx: In function 'int itkOrientImageFilterTest(int, char**)':
/home/matt/src/ITK/Modules/Filtering/ImageGrid/test/itkOrientImageFilterTest.cxx:85:42: warning: 'itk::AnatomicalOrientation::AnatomicalOrientation(LegacyOrientationType)' is deprecated: Use the AnatomicalOrientation::FromEnum type instead. [-Wdeprecated-declarations]
   85 |   orienter->SetGivenCoordinateOrientation(
      |   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
   86 |     itk::SpatialOrientationEnums::ValidCoordinateOrientations::ITK_COORDINATE_ORIENTATION_RIP);
      |     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /home/matt/src/ITK/Modules/Filtering/ImageGrid/include/itkOrientImageFilter.h:23,
                 from /home/matt/src/ITK/Modules/Filtering/ImageGrid/test/itkOrientImageFilterTest.cxx:19:
/home/matt/src/ITK/Modules/Core/Common/include/itkAnatomicalOrientation.h:444:3: note: declared here
  444 |   AnatomicalOrientation(LegacyOrientationType legacyOrientation);
      |   ^~~~~~~~~~~~~~~~~~~~~
/home/matt/src/ITK/Modules/Filtering/ImageGrid/test/itkOrientImageFilterTest.cxx:90:44: warning: 'itk::AnatomicalOrientation::AnatomicalOrientation(LegacyOrientationType)' is deprecated: Use the AnatomicalOrientation::FromEnum type instead. [-Wdeprecated-declarations]
   90 |   orienter->SetDesiredCoordinateOrientation(
      |   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
   91 |     itk::SpatialOrientationEnums::ValidCoordinateOrientations::ITK_COORDINATE_ORIENTATION_IRP);
      |     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/matt/src/ITK/Modules/Core/Common/include/itkAnatomicalOrientation.h:444:3: note: declared here
  444 |   AnatomicalOrientation(LegacyOrientationType legacyOrientation);
      |   ^~~~~~~~~~~~~~~~~~~~~
/home/matt/src/ITK/Modules/Filtering/ImageGrid/test/itkOrientImageFilterTest.cxx:126:42: warning: 'itk::AnatomicalOrientation::AnatomicalOrientation(LegacyOrientationType)' is deprecated: Use the AnatomicalOrientation::FromEnum type instead. [-Wdeprecated-declarations]
  126 |   orienter->SetGivenCoordinateOrientation(
      |   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
  127 |     itk::SpatialOrientationEnums::ValidCoordinateOrientations::ITK_COORDINATE_ORIENTATION_RIP);
      |     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/matt/src/ITK/Modules/Core/Common/include/itkAnatomicalOrientation.h:444:3: note: declared here
  444 |   AnatomicalOrientation(LegacyOrientationType legacyOrientation);
      |   ^~~~~~~~~~~~~~~~~~~~~
/home/matt/src/ITK/Modules/Filtering/ImageGrid/test/itkOrientImageFilterTest.cxx:128:44: warning: 'itk::AnatomicalOrientation::AnatomicalOrientation(LegacyOrientationType)' is deprecated: Use the AnatomicalOrientation::FromEnum type instead. [-Wdeprecated-declarations]
  128 |   orienter->SetDesiredCoordinateOrientation(
      |   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
  129 |     itk::SpatialOrientationEnums::ValidCoordinateOrientations::ITK_COORDINATE_ORIENTATION_LIP);
      |     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/matt/src/ITK/Modules/Core/Common/include/itkAnatomicalOrientation.h:444:3: note: declared here
  444 |   AnatomicalOrientation(LegacyOrientationType legacyOrientation);
```

and this test failure message:

```
3056:   File "/home/matt/src/ITK/build-python/Wrapping/Generators/Python/itk/itkOrientImageFilterPython.py", line 1530, in SetDesiredCoordinateOrientation
3056:     return _itkOrientImageFilterPython.itkOrientImageFilterIUC3IUC3_SetDesiredCoordinateOrientation(self, newCode)
3056:            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
3056: TypeError: in method 'itkOrientImageFilterIUC3IUC3_SetDesiredCoordinateOrientation', argument 2 of type 'itkAnatomicalOrientation'
```

So it seems we need to either

a) Update the of OrientImageFilter to use AnatomicalOrientation (preferred?) or
b) Have the test use the old enum

Reply:
> a) Update the of OrientImageFilter to use AnatomicalOrientation (preferred?) 

Yes, this PR does this update. It appears my latest change has the Python testing. It appears that for Python the implicit conversion from the enum type to AnatomicalOrientation is needed. The implicit constructors for AnatomicalOrientation do not seem to implicitly do the conversion.

> b) Have the test use the old enum
The conversion from the old enum to AnatomicalOrientation is not implicit in Python. Additionally in C++ the conversion is marked FUTURE_LEGACY_REMOVE and difficult for testing.